.gitlab/ci/global.yml

```diff
@@ -13,66 +13,235 @@
+  - section_start "Build Parameters"
+  - set | grep -E "^NIGHTLY=|^INSTANCE_TESTS=|^SERVER_BRANCH_NAME=|^ARTIFACT_BUILD_NUM=|^DEMISTO_SDK_NIGHTLY=|^TIME_TO_LIVE=|^CONTRIB_BRANCH=|^FORCE_PACK_UPLOAD=|^PACKS_TO_UPLOAD=|^BUCKET_UPLOAD=|^GCS_MARKET_BUCKET=|^SLACK_CHANNEL="
+  - python --version
+  - python3 --version
+  - node --version
+  - npm --version
+  - demisto-sdk --version
+  - section_end "Build Parameters"
+
+.default-before_script:
+  before_script: *prepare_env
+
+.default-after-script:
+  after_script:
+    - *prepare_env
+    - section_start "Notify Slack"
     - |
-      if [ -f "./venv/bin/activate" ]; then
-        echo "found venv"
+      if [ -n "${SLACK_ONLY_IF}" ]; then
+        ./Tests/scripts/slack_notifier.sh "$SLACK_TEST_TYPE" "$SLACK_ENV_RESULTS" "$SLACK_CHANNEL"
       else
-        echo "installing venv"
-        NO_HOOKS=1 SETUP_PY2=yes .hooks/bootstrap >> $ARTIFACTS_FOLDER/logs/installations.log 2>&1
+        echo "Skipping Slack"
       fi
-    - venv/bin/pip3 install -r .circleci/build-requirements.txt >> $ARTIFACTS_FOLDER/logs/installations.log 2>&1
-    - source ./venv/bin/activate
+    - section_end "Notify Slack"
+    - section_start "Destroy instances"
     - |
-      if [ -n "${DEMISTO_SDK_NIGHTLY}" ]; then
-        echo "Installing SDK from master branch" | tee --append $ARTIFACTS_FOLDER/logs/installations.log
-        pip3 install git+https://github.com/demisto/demisto-sdk@master >> $ARTIFACTS_FOLDER/logs/installations.log 2>&1
+      if [ -n "${DESTROY_INSTANCES}" ]; then
+        python3 ./Tests/scripts/destroy_instances.py $ARTIFACTS_FOLDER $ARTIFACTS_FOLDER/env_results.json "$INSTANCE_ROLE" "$TIME_TO_LIVE"
       fi
-    - section_end "Installing Virtualenv"
-
-    - section_start "Installing SSH keys" --collapsed
-    - eval $(ssh-agent -s)
-    - chmod 400 $OREGON_CI_KEY
-    - ssh-add $OREGON_CI_KEY
-    - mkdir -p ~/.ssh
-    - chmod 700 ~/.ssh
-    - section_end "Installing SSH keys"
-
-    - section_start "Build Parameters"
-    - set | grep -E "^NIGHTLY=|^INSTANCE_TESTS=|^SERVER_BRANCH_NAME=|^ARTIFACT_BUILD_NUM=|^DEMISTO_SDK_NIGHTLY=|^TIME_TO_LIVE=|^CONTRIB_BRANCH=|^FORCE_PACK_UPLOAD=|^PACKS_TO_UPLOAD=|^BUCKET_UPLOAD=|^GCS_MARKET_BUCKET=|^SLACK_CHANNEL="
-    - python --version
-    - python3 --version
-    - node --version
-    - npm --version
-    - demisto-sdk --version
-    - section_end "Build Parameters"
-
+    - section_end "Destroy instances"
 
 .default-job-settings:
   extends:
     - .default-cache
     - .default-before_script
+    - .default-after-script
+
+.run-unittests-and-lint:
+  needs: []
+  stage: unittests-and-validations
+  artifacts:
+    expire_in: 30 days
+    paths:
+      - /builds/xsoar/content/unit-tests
+      - /builds/xsoar/content/artifacts/*
+  extends:
+    - .default-job-settings
+  variables:
+    SLACK_TEST_TYPE: "unittests"
+    SLACK_ENV_RESULTS: $ARTIFACTS_FOLDER/env_results.json
+    SLACK_ONLY_IF: $BUCKET_UPLOAD
```

Comment:
Seems odd to have these environment variable defaults for the hidden job (well more specifically using `BUCKET_UPLOAD` for `SLACK_ONLY_IF`.